src/main/java/com/artipie/Pie.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 artipie.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.artipie;
+
+import com.artipie.asto.Key;
+import com.artipie.asto.Storage;
+import com.artipie.http.Connection;
+import com.artipie.http.Response;
+import com.artipie.http.Slice;
+import com.artipie.http.rq.RequestLineFrom;
+import com.artipie.http.rs.RsWithStatus;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.ByteBuffer;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Flow.Publisher;
+
+/**
+ * Pie of slices.
+ * @since 1.0
+ * @todo #12:30min Implement slice resolving strategy
+ *  based on yaml configuration file. Now SliceStub
+ *  is used instead of real slice implementation.
+ *  We should parse publisher of bytes into yaml
+ *  config, construct ASTO from this config and find
+ *  corresponding slice implementation by type parameter.
+ * @checkstyle MagicNumberCheck (500 lines)
+ * @checkstyle ReturnCountCheck (500 lines)
+ */
+public final class Pie implements Slice {
+
+    /**
+     * Configuration storage.
```

Comment:
Improve the description, please. Mention repository configuration files.

Reply:
@Sammers21 done, see https://github.com/artipie/artipie/pull/18/commits/632252f58f80ab903bbbbee9fe9cb4987a7c7ee3